src/client/executor.rs

```diff
@@ -106,8 +130,23 @@ impl Client {
             .execute_operation_on_connection(&op, &mut conn, &mut session, txn_number)
             .await
         {
-            Ok(result) => {
-                return Ok(result);
+            Ok(OperationResponse {
```

Comment:
I personally prefer this match with the condition checking in the branch like this:
```rust
match blah {
   Ok(response) => {
        let connection = if response.more_to_come { Some(conn) } else { None };
        return Ok(OperationResult { response: response.response, connection })
    },
    Err(err) => { ... }
}
```
This way the duplicated `OperationResponse { ... }` matching can be omitted. Up to you if you think it's worth changing.